### DISCUSSION TITLE
首届“iLogtail开源直通车”活动圆满落幕！

### DISCUSSION THREAD

**Discussion by henryzhx8** [alibaba/ilogtail#1239] [Announcements]
经过4个多月的努力，首届“iLogtail开源直通车”活动即将进入尾声。在这4个月的时间内，我们总共收到了20多位来自全国各地在校学生的申请，其中超半数的同学完整地参与了整个活动的全部内容。不论是采集配置的编写，项目的设计，还是代码的编写与测试，各位同学的热情、好学以及不耻下问很好地诠释了开源背后所蕴含的意义。

### 活动回顾

整个“iLogtail开源直通车“活动共经历了3个阶段，接下来就让我们一起来简单回顾一下～

- 活动预热

iLogtail作为一款可观测数据采集器，对于广大在校学生而言相对比较陌生。为了让大家能够快速上手开发，同时对iLogtail有一个直观的认识，我们鼓励参与活动的同学寻找自己学习工作中遇到过的日志，并运用iLogtail现有的处理插件对日志进行处理。在此倡议下，同学们踊跃贡献配置模版，其中不乏一些普适性较强的例子，例如：Redis-cli日志（张杨）、rsync日志（马晓龙）、tcpdump日志（金煜峰）等，为社区其他用户使用iLogtail解析日志提供了巨大的便利性。

- 项目设计

本次活动设立了3个不同维度和不同难度等级的项目，包括“过滤插件易用性提升”（项目1）、“特定格式日志解析性能比拼”（项目2）、“编写SQL处理插件”（项目3），难度等级依次上升。在项目设计阶段，同学们都大开脑洞，展现了十八般武艺，例如：项目1的同学提出了基于shell脚本语言的解析器，项目3的同学对于词法分析的优化提出了自己的独到见解等。除了设计思路的新颖性，大部分同学的设计也都考虑了设计的缜密性，对包括边界条件在内的特殊场景做了一定的处理。从各个维度而言，大部分同学的设计都呈现出较高的质量，我们也对设计中的不足进行了及时的指点，帮助同学们更好地完善设计。

- 项目开发

为了让尽可能多的同学参与到开源中来，所有提交设计的同学都被给予了将设计落地的机会，并且大部分同学都最终完成了代码的开发。尽管每个项目最终只有1位同学的代码有机会合入主干，但同学们依然以高标准完成了代码的编写工作，部分同学甚至提供了单元测试和集成测试，一定程度提升了代码的可靠性。对于最终计划合入主干的代码，我们也进行了一对一的指导，帮助同学提升工程代码的质量。

### 获奖名单

结合各位同学的活动参与情况，以及项目难度和项目完成质量等因素，我们最终评选出本次活动的优胜者，名单如下：

- 一等奖：邱风硕

- 二等奖：胡敬松、蒋伟博

- 三等奖：汪嘉华、金煜峰、李润洋

恭喜以上获奖同学，我们会为获奖同学送上精美奖品！🎉🎉🎉

<p align="center">
  <img src="https://github.com/alibaba/ilogtail/assets/38356684/383955db-b6e3-4c47-91ca-c66081034c4f" width="400">
</p>

### 优秀案例点评

项目3是本次活动中难度最大的项目，需要同学们熟悉 iLogtail 的插件框架，并具备丰富的编译原理和数据库原理知识。尽管面临挑战，邱风硕同学依然展现出了出色的能力和才华，并且在有限的时间内提交了功能完整的代码。在项目初期的设计阶段，该生就对词法分析、语法分析和语义分析等阶段进行了详细的设计，同时根据项目的特点提出了各自的裁剪和优化方法。这些设计思路在后续的实现中得到了很好的体现。得益于精心的设计规划，提交的代码在执行效率上表现出色，不仅在代码功能方面做得出色，并且代码结构组织清晰、异常处理得当，附带了完善的单元测试，这反映了邱风硕同学对于保障代码质量方面的意识。通过参与该项目，学生可提升自己的编译原理和数据库原理的能力，同时也能够熟悉和掌握 iLogtail 的插件开发，并为开源社区的发展贡献自己的力量。
—— by 导师：余韬

### 关于iLogtail
iLogtail是一款高性能的轻量级可观测数据采集器，由阿里云SLS团队官方提供，可以运行在包括服务器、容器和嵌入式等多种环境中，其宗旨在于帮助开发者构建统一的数据采集层，助力可观测平台打造各种上层应用场景。iLogtail多年来一直稳定服务阿里集团、蚂蚁集团以及众多公有云上的企业客户，目前已经有千万级的安装量，每天采集数十PB的可观测数据，广泛应用于线上监控、问题分析/定位、运营分析、安全分析等多种场景。

**Comment by YinXian-Cheng** [comment]
恭喜邱风硕同学
